### PR TITLE
support platforms with no os.link

### PR DESCRIPTION
**Pull request by ThomasWaldmann**

### CODE REVIEW

File: src/borg/archiver.py [hunk 1]
```diff
@@ -744,8 +744,17 @@ def do_extract(self, args, repository, manifest, key, archive):
         hardlink_masters = {} if partial_extract else None
```
Comment (willsALMANJ) [inline src/borg/archiver.py]: This needs to be something like:

```
         hardlink_masters = {} if partial_extract or not has_link else None 
```

Comment (ThomasWaldmann) [reply src/borg/archiver.py]: I'll fix this, thanks!

Comment (ThomasWaldmann) [reply src/borg/archiver.py]: fixed.

do you think it is ready to merge now?

Comment (ThomasWaldmann) [reply src/borg/archiver.py]: update: reread the ticket, answer seems "yes".

**Pull request merged**